pennylane/circuit_graph.py

```diff
@@ -86,83 +88,38 @@ class CircuitGraph:
         ops (Iterable[.Operator]): quantum operators constituting the circuit, in temporal order
         obs (Iterable[.MeasurementProcess]): terminal measurements, in temporal order
         wires (.Wires): The addressable wire registers of the device that will be executing this graph
-        par_info (list[dict]): Parameter information. For each index, the entry is a dictionary containing an operation
+        par_info (Optional[list[dict]]): Parameter information. For each index, the entry is a dictionary containing an operation
         and an index into that operation's parameters.
-        trainable_params (set[int]): A set containing the indices of parameters that support
+        trainable_params (Optional[set[int]]): A set containing the indices of parameters that support
             differentiability. The indices provided match the order of appearence in the
             quantum circuit.
     """
 
-    # pylint: disable=too-many-public-methods
-
-    def __init__(self, ops, obs, wires, par_info=None, trainable_params=None):
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        ops: list[Union[Operator, MeasurementProcess]],
+        obs: list[MeasurementProcess],
+        wires: Wires,
+        par_info: Optional[list[dict]] = None,
+        trainable_params: Optional[set[int]] = None,
+    ):
         self._operations = ops
         self._observables = obs
         self.par_info = par_info
         self.trainable_params = trainable_params
 
-        queue = ops + obs
+        self._queue = ops + obs
 
-        self._depth = None
-
-        self._grid = {}
-        """dict[int, list[Operator]]: dictionary representing the quantum circuit as a grid.
-        Here, the key is the wire number, and the value is a list containing the operators on that wire.
-        """
-
-        self._indices = {}
-        # Store indices for the nodes of the DAG here
-
-        self.wires = wires
+        self.wires = Wires(wires)
         """Wires: wires that are addressed in the operations.
         Required to translate between wires and indices of the wires on the device."""
         self.num_wires = len(wires)
         """int: number of wires the circuit contains"""
-        for k, op in enumerate(queue):
-            # meas_wires = wires or None  # cannot use empty wire list in MeasurementProcess
-            op.queue_idx = k  # store the queue index in the Operator
```

Comment:
yes this is no longer updated.  We should be able to remove it from `Operator` now.

Reply:
Tagging @EmilianoG-byte as it will make his work to remove `op.queue_idx` much easier.

Reply:
This is the reason I slated `Opeator.queue_idx` for removal this cycle.

Reply:
Ohhh I see. I was already starting to work on how to get rid of `queue_idx` and Mudit and I were discussing exactly this. I should have asked you before @albi3ro